src/compiler/compile-props.js

```diff
@@ -229,10 +229,10 @@ function makePropsLinkFn (props) {
 
 function processPropValue (vm, prop, rawValue, fn) {
   const isSimple = prop.dynamic && isSimplePath(prop.parentPath)
-  let value = coerceProp(prop, rawValue)
-  if (value === undefined) {
-    value = getPropDefaultValue(vm, prop)
+  if (rawValue === undefined) {
+    rawValue = getPropDefaultValue(vm, prop)
   }
+  let value = coerceProp(prop, rawValue)
```

Comment:
If you want to keep compatibility with the old version you'll need to check again if the value is undefined, althought I don't see why your coerce fn would return undefined...